app/controllers/users/registrations_controller.rb

```diff
@@ -8,7 +8,7 @@ def create_user_from_invitation
     @user.email = @invitation.email
     if @user.save
       sign_in @user
-      redirect_to accept_invitation_path(token: @invitation.token)
+      redirect_to accept_invitation_path(@invitation, token: @invitation.token)
```

Comment:
@Gataniel you should not use `id` in invitation route, so user can't find foreign invitations just by changing `id`. You should always use token. So in this case you may just pass token as id:
```rb
accept_invitation_path(id: @invitation.token)
```
and in controller
```rb
def find_invitation
  Invitation.find_by!(token: params[:id])
end
```

Or declare a new route with token:
```rb
get `/invitation/:token/accept` => `invitations#accept`
```